### DISCUSSION TITLE
FieldsService.readOne throws Forbidden if field not exists

### DISCUSSION THREAD

**Discussion by rokdd** [directus/directus#15295] [Feature Requests]
### Describe the Bug

If I query with readOne for not defined field in a collection I get the messages in the errors section copied thrown as Exception. I would expect a return value of false, null,..

### To Reproduce

Works:
`this.fields_service.readOne("collection","date_updated")`
Does not work:
`this.fields_service.readOne("collection","date_released")`

### Errors Shown

```
ForbiddenException [Error]: You don't have permission to access this.
    at FieldsService.readOne (C:\......\node_modules\directus\dist\services\fields.js:203:19)      
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.check_config (C:\......\extensions\endpoints\t\index.js:420:17)
    at async C:\.....\extensions\endpoints\t\index.js:1219:11 {
  status: 403,
  code: 'FORBIDDEN',
  extensions: {}
}
```

**Comment by azrikahar** [comment]
In order to fix #9366, non existent fields & collections were intentionally aligned with how non existent relations behaves in #12952.

Ideally it should not be the case when you're explicitly authenticated as an admin, but that's currently being discussed in #4368 as mentioned in this comment: https://github.com/directus/directus/issues/9366#issuecomment-958484522 👍

**Comment by rokdd** [comment]
Well thank you, I understand the sense behind (y). I was auth as admin, so theoretically I should not see any Forbidden?! However I would say an exist function would make sense for the field service, but that is my personal one. Here is my function which I created and works now:
```
async function exists_field (req, collection, name)
        {
            let fs=new services.FieldsService({ schema: req.schema, accountability: req.accountability });
            try {
                await fs.readOne(collection, name)
            }
            catch
            {
                return false
            }
            return true
        },
```

**Reply by rijkvanzanten** [reply]
> I was auth as admin, so theoretically I should not see any Forbidden?!

Kinda. As of right now, Directus will return forbidden for wrong field names and IDs as well, to ensure that users (authenticated or otherwise) can't extract what items/fields exist within the API. It also makes sure that the same error codes are returned regardless of user for consistency's sake. There's an open discussion debating adding a way to set the "friendliness" of the errors here: https://github.com/directus/directus/discussions/4368.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
